src/job.rs

```diff
@@ -390,9 +390,12 @@ pub fn perform_imap_fetch(context: &Context) {
     if 0 == connect_to_inbox(context, &inbox) {
         return;
     }
-    if !context.get_config_bool(Config::InboxWatch) {
-        info!(context, "INBOX-watch disabled.",);
-        return;
+    match context.get_config_item(ConfigKey::InboxWatch) {
+        Some(ConfigItem::InboxWatch(false)) => {
+            info!(context, "INBOX-watch disabled.",);
+            return;
+        }
+        _ => (),
```

Comment:
i think, here is a sight behavior change:

on bad config-key, the old implementation aborts, the new one continues (wondering if the latter can happen at all due to default values and strict key names)

howver, probably fine in this case, however, i think there is some potential to break things :)

Reply:
Great catch!  Yes, this should handle the None case like the Some(false) case.

Reply:
Also converted to `if let` which probably makes more sense.